index.php

```diff
@@ -38,14 +38,24 @@
 $languages = array();
 if ($handle = opendir('lang')) {
     while (false !== ($entry = readdir($handle))) {
-        if ($entry != "." && $entry != "..") {
-             array_push($languages, str_replace(".inc.php", "", $entry));
+        if ($entry != "." && $entry != ".." ) {
+            $entry_lang = str_replace(".inc.php", "", $entry);
+            # Only add language to possibilities if it is the default language or part of the allowed languages 
+            # empty $allowed_lang <=> all languages are allowed
+            if ($entry_lang == $lang || empty($allowed_lang) || in_array($entry_lang, $allowed_lang) ) {
+                array_push($languages, $entry_lang);
+            }
         }
     }
     closedir($handle);
 }
 $lang = detectLanguage($lang, $languages);
-require_once("lang/$lang.inc.php");
+if (file_exists("lang/$lang.inc.php")) {
```

Comment:
I do not see a value in this if / else block.

$lang here is either the default $lang from the config file or one from the $languages array.
All files with $lang from $languages exist because they are only added to in the while loop reading the files inside the _./lang_ dir.

So the only way for this expression to return false is that the default $lang from config file has no associated lang/...inc.php file. That's probably a typo e.g. `$lang = "En";` or the associated language file does not exist.
In all cases, there is a configuration error. I would rather log this e.g "please fix $lang from config" or leave the `require_once("lang/$lang.inc.php");` so PHP will halt the script and tell sorry there is a missing file.